internals/secrethub/read_test.go

```diff
@@ -1,9 +1,127 @@
 package secrethub
 
 import (
+	"io/ioutil"
+	"os"
 	"testing"
+	"time"
+
+	"github.com/secrethub/secrethub-cli/internals/cli/clip/fakeclip"
+	"github.com/secrethub/secrethub-cli/internals/cli/filemode"
+	"github.com/secrethub/secrethub-cli/internals/cli/ui/fakeui"
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/internals/assert"
+	"github.com/secrethub/secrethub-go/internals/errio"
+	"github.com/secrethub/secrethub-go/pkg/secrethub"
+	"github.com/secrethub/secrethub-go/pkg/secrethub/fakeclient"
 )
 
 func TestReadCommand_Run(t *testing.T) {
-	// TODO SHDEV-1029 Test ReadCommand.
+	testErr := errio.Namespace("test").Code("test").Error("test error")
+	testSecret := []byte("Remember! Reality's an illusion, the universe is a hologram, buy gold! Bye! - Bill Cipher")
+
+	cases := map[string]struct {
+		cmd             ReadCommand
+		newClientErr    error
+		secretVersion   api.SecretVersion
+		serviceErr      error
+		expectedClip    []byte
+		expectedFileOut []byte
+		out             string
```

Comment:
For consistency, consider naming this field `expectedOut` or `expectedStdout`, whichever makes the most sense. Naming it `out` in this sense makes it look like a parameter, but it's actually part of the assertion. So matching the `actual` and `expected` pattern makes for more intuitive code.